### PR TITLE
Event access by url and parsing issues solved.

### PR DESCRIPTION
**Pull request by materiywakal**

### CODE REVIEW

File: src/ReKreator/ReKreator.BL/Services/EventHandler.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    public class EventHandler
+    {
+        private IUnitOfWork _unitOfWork;
+
+        public EventHandler(IUnitOfWork unitOfWork)
+        {
+            _unitOfWork = unitOfWork;
+        }
+
+        public async Task<Event> GetEventByUrl(string sourceUrl)
```
Comment (ArBond) [inline src/ReKreator/ReKreator.BL/Services/EventHandler.cs]: ...Async

File: src/ReKreator/ReKreator.UI.MVC/Views/Home/Index.cshtml [hunk 2]
```diff
@@ -69,7 +69,7 @@
                                         <span class="aa-price">
                                             Event ends at @value.ExpiryDate.ToString("dd.MM.yyyy")
```
Comment (alexfromny) [inline src/ReKreator/ReKreator.UI.MVC/Views/Home/Index.cshtml]: we have a project date format, please use that

File: src/ReKreator/ReKreator.UI.MVC/Views/Home/Index.cshtml [hunk 3]
```diff
@@ -22,7 +22,7 @@
                         <div class="aa-top-slider-content">
                             <span class="aa-top-slider-catg">@view.Title</span>
                             <p class="aa-top-slider-location"><i class="fa fa-map-marker"></i>@viewdata.EventPlace.Title</p>
-                            <span class="aa-top-slider-off">@viewdata.Time.ToString("dd.MM.yyyy")</span>
+                            <span class="aa-top-slider-off">@viewdata.Time.ToString(ReKreator.Parsing.ParserUtilities.DateTimeFormat)</span>
```
Comment (alexfromny) [inline src/ReKreator/ReKreator.UI.MVC/Views/Home/Index.cshtml]: we have DateTimeFormats.ShortDateFormat

File: src/ReKreator/ReKreator.UI.MVC/Controllers/EventController.cs [hunk 4]
```diff
@@ -19,16 +20,25 @@ public class EventController : Controller
...
+        public async Task<IActionResult> Event(string sourceUrl)
         {
-            return View();
+            var item = await _service.GetEventByUrlAsync(sourceUrl);
+            if (item == null)
+            {
+                return RedirectToAction("Error404", "Error");
+            }
+            var viewItem = _mapper.Map<EventViewModel>(item);
+            viewItem.ShortUrl = viewItem.SourceUrl.Split("/").SkipLast(1).Last();
```
Comment (alexfromny) [inline src/ReKreator/ReKreator.UI.MVC/Controllers/EventController.cs]: Are you sure that all urls will end with "/"?

Comment (materiywakal) [reply src/ReKreator/ReKreator.UI.MVC/Controllers/EventController.cs]: Yes!

**Pull request merged**